lib/system/gc_common.nim

```diff
@@ -144,15 +144,35 @@ when allowForeignThreadGc:
     ## This function is available only when ``--threads:on`` and ``--tlsEmulation:off``
     ## switches are used
     if not localGcInitialized:
+      if not gch.stackBottom.isNil:
```

Comment:
well if you call it via callback, then this mechanism should wrap it in setup/tearDown.